stubs/setuptools/setuptools/__init__.pyi

```diff
@@ -45,6 +45,13 @@ __all__ = [
 
 __version__: str
 
+class _Library(TypedDict):
```

Comment:
Should this be marked `@type_check_only`? There seems to be some precedence for using it on some similar typeddicts, e.g. https://github.com/python/typeshed/blob/7ccc587afb50d3b7f77628292a8136b4c1e70735/stdlib/_locale.pyi#L5-L6

Also, to bikeshed the name, I see setuptools internally calls this structure a `build_info` in a few places, so maybe `_BuildInfo` would be a good fit?

Reply:
Interesting, I wasn't aware of it, but likely what we want here. Thanks for the callout. Agree that's a better name here.

Reply:
https://github.com/python/typeshed/pull/12791/commits/f0d929f3f2a8767f56bfffd953cc3fb0f923e133